pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/thread/ThreadRuntime.java

```diff
@@ -145,6 +204,12 @@ public void stop() {
             }
             // make sure JavaInstanceRunnable is closed
             this.javaInstanceRunnable.close();
+
+            // once the thread quits, clean up the instance
+            fnCache.unregisterFunctionInstance(
+                    instanceConfig.getFunctionId(),
+                    instanceConfig.getInstanceName());
+            log.info("Unloading JAR files for function {}", instanceConfig);
```

Comment:
Do we have to move this line one line above?

Reply:
sure